### PR TITLE
Add managed single accordion component

### PR DESCRIPTION
**Pull request by votemike**
We should leave the ReactAccordion component roughly as is so that people can manage expanding collapsing as they see fit. But it would be good to introduce an option for them to have the expanding/collapsing managed for them.

Is this the way forward we want to go?

Should we update the README with instructions for this yet? Or do the README in a big bang?

Anything else need changing/adding?

### CODE REVIEW

**Comment by markwylde** [comment]
Fixed the merge conflicts for you, and added both varients to the demo page.

Also had a look, and can see the issue. Not sure why, but I'll keep looking.

**Comment by markwylde** [comment]
Okay I think I see the issue. The browser is adding the tag automatically before your onClick runs. So, by the time your logic runs, it's already open.

Try this, and it works.

```diff
diff --git a/src/index.tsx b/src/index.tsx
index 5316f43..f6971d0 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -13,14 +13,16 @@ interface ReactAccordionProps {
 export function ReactAccordion(props: ReactAccordionProps) {
   const { items } = props;
 
-  const handleClick = (index) => () => props.onClick && props.onClick(index);
+  const handleClick = (index) => () => {
+    setTimeout(() => props.onClick && props.onClick(index))
+  };
 
   return (
     <ul>
       {items.map((item, index) => (
         <li key={item.name}>
           <details aria-label={item.name} open={item.open} onClick={handleClick(index)}>
-            <summary>{item.heading}</summary>
+            <summary>{item.heading}           {item.open?.toString()}</summary>
             {item.child}
           </details>
         </li>

```

File: demo/index.tsx [hunk 1]
```diff
@@ -1,13 +1,18 @@
 import React from 'react';
```
Comment (votemike) [inline demo/index.tsx:1]: What is this file compared to demo/src/index.tsx?

File: package.json [hunk 2]
```diff
@@ -41,13 +41,16 @@
...
+    "@typescript-eslint/parser": "^5.38.1",
     "eslint-config-airbnb-typescript": "^17.0.0",
-    "jest": "^29.0.3",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint": "^8.24.0",
     "jest-environment-jsdom": "^29.0.3",
+    "jest": "^29.0.3",
+    "react-dom": "^18.2.0",
+    "servatron": "^2.3.2",
```
Comment (votemike) [inline package.json:53]: Do I need this? Was this part of a bad merge conflict?

Comment (markwylde) [reply package.json:53]: Yeah this looks like a merge conflict thing. It's not needed. Neither is `react-dom`.

File: demo/index.tsx (hunk 1 above)
Comment (markwylde) [reply demo/index.tsx:1]: It's old I think. We can [delete it](https://github.com/votemike/react-accordion/pull/6/files).

File: src/__tests__/index.tsx [hunk 3]
```diff
@@ -37,9 +37,28 @@ test('an accordion item details has an aria-name', () => {
 
 test('the open prop is passed to the item', () => {
```
Comment (votemike) [inline src/__tests__/index.tsx:38]: This test seems to be wrong.... we need to fix it in a bit.

File: src/index.tsx [hunk 4]
```diff
@@ -26,5 +30,25 @@ export function ReactAccordion(props: ReactAccordionProps) {
   );
 }
 
+export function SingleItemOpenAccordion(props: { items: Item[] }) {
```
Comment (votemike) [inline src/index.tsx:33]: What if someone passes 2 items initially with the open prop?

File: src/index.tsx [hunk 5]
```diff
@@ -26,5 +30,25 @@ export function ReactAccordion(props: ReactAccordionProps) {
...
+  }));
+
+  return (
+    <ReactAccordion items={revisedItems} onClick={handleClick} />
+  );
+}
+
 export default ReactAccordion;
 module.exports = ReactAccordion;
+module.exports.SingleItemOpenAccordion = SingleItemOpenAccordion;
```
Comment (votemike) [inline src/index.tsx:54]: Is this the best way to export these?

Comment (markwylde) [reply src/index.tsx:54]: Almost definitely not, but I'm not sure of an alternative/better way to handle commonjs support.

File: demo/dist/index.html [hunk 6]
```diff
@@ -1,14 +0,0 @@
-<!DOCTYPE html>
```
Comment (votemike) [inline demo/dist/index.html:1]: Are we supposed to keep this file?

File: src/__tests__/index.tsx (hunk 3 above)
Comment (votemike) [reply src/__tests__/index.tsx:38]: https://github.com/votemike/react-accordion/pull/7

File: demo/dist/index.html (hunk 6 above)
Comment (markwylde) [reply demo/dist/index.html:1]: Yeah. I think eventually we'll add a build script that will copy this from the `src` directory?

File: demo/index.tsx [hunk 7]
```diff
@@ -1,31 +0,0 @@
-import React from 'react';
```
Comment (votemike) [inline demo/index.tsx:1]: Does this need to be kept?

Comment (markwylde) [reply demo/index.tsx]: It looks like it's not needed. Though I think we need to update the eslint rules.

File: demo/src/index.tsx [hunk 8]
```diff
@@ -13,12 +13,20 @@ const data = [
...
   }
 ];
 
 const DemoPage = () => {
   return (
-    <ReactAccordion items={data}/>
+    <>
+      <ReactAccordion items={data} />
+      <SingleItemOpenAccordion items={data} />
+    </>
```
Comment (cat-markwylde) [inline demo/src/index.tsx:29]: Can we give some headings to the demo page?

```suggestion
    <>
      <h1>ReactAccordion Demo</h1>
      <h2>Accordion that allows multiple open</h2>
      <ReactAccordion items={data} />
      
      <h2>Accordion that only allows one open</h2>
      <SingleItemOpenAccordion items={data} />
    </>
```

File: src/index.tsx (hunk 4 above)
Comment (cat-markwylde) [reply src/index.tsx:33]: I would think the last one should work, and the others should be ignored?

**Review by cat-markwylde — APPROVED** [review]

File: src/index.tsx (hunk 4 above)
Comment (votemike) [reply src/index.tsx:33]: Updated.
What do you think?
Best way to do it?

Comment (markwylde) [reply src/index.tsx:33]: Ahh cool. I like the reverse!

**Review by markwylde — APPROVED** [review]

**Pull request merged**